### PR TITLE
Automatically host preview version of docs for PRs

### PR DESCRIPTION
**Pull request by geel9**
This PR creates a new GitHub action which runs on every PR that touches docs; it builds that version of the docs and hosts a static site in netlify.

### CODE REVIEW

File: .github/workflows/test-docs.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+name: Documentation (PR preview)
+
+on:
+  pull_request:
+    paths:
+      - "docs/**"
```
Comment (fundthmcalculus) [inline .github/workflows/test-docs.yml]: Docs/**/*

**Review by fundthmcalculus — CHANGES_REQUESTED** [review]
See comment. Also, requesting changes to block merge. Will approve when done.

**Comment by janpieterz** [comment]
Interesting. Do we use Netlify for our normal docs?

**Comment by geel9** [comment]
> Interesting. Do we use Netlify for our normal docs?

No -- we use GitHub Pages.

The normal docs build process commits to the `gh-pages` branch of this repo, which is served from our docs site.

However, there is no way to have more than one branch of GitHub pages -- we'd need to either make a separate repository, or do some other kind of hackery.

Initially I wanted to use Azure Storage for this, which works fine when we're just deploying a single static site, but deploying multiple static sites is... tricky. 

That is -- with Azure, if we deployed a single static site to the root directory of a storage container, it works just fine, but to support multiple PRs having previews available concurrently, we'd have to place each branch into a subdirectory of said container, which breaks when relative URLs are at play on the docs site (which is in a lot of places)


The Azure solution would be to create a new storage unit for each branch, configure it appropriately, generate an access token for it, and then deploy the site to it.

I'm A) not familiar enough with Azure and B) not confident that this is a sane solution.



So, I searched for a service which is designed to programmatically host static sites, and Netlify seems like it's going to be a good fit. I think we'll even be able to use their free tier indefinitely for this purpose.

File: .github/workflows/test-docs.yml (hunk 1 above)
Comment (geel9) [reply .github/workflows/test-docs.yml]: @fundthmcalculus Ready for re-review

**Review by fundthmcalculus — APPROVED** [review]
Nice!! @janpieterz

**Pull request merged**